components/tooltip/tooltip.js

```diff
@@ -931,25 +946,49 @@ class Tooltip extends RtlMixin(LitElement) {
 		this.showing = this._isFocusing || this._isHovering || this.forceShow;
 	}
 
-	_updateTarget() {
+	async _updateTarget() {
 		this._removeListeners();
-		this._target = this._findTarget();
+
+		if (this._target && this.id) {
+			elemIdListRemove(this._target, 'aria-labelledby', this.id);
+			elemIdListRemove(this._target, 'aria-describedby', this.id);
+			if (this.#targetDelegated) {
+				this._target.removeAttribute('aria-label');
+				this._target.removeAttribute('aria-description');
+			}
+		}
+
+		this._target = await this._findTarget();
 		if (this._target) {
 			const isInteractive = this._isInteractive(this._target);
 			this.id = this.id || getUniqueId();
 			this.setAttribute('role', 'tooltip');
-			if (this.forType === 'label') {
-				elemIdListAdd(this._target, 'aria-labelledby', this.id);
-			} else if (!this.announced || isInteractive) {
-				elemIdListAdd(this._target, 'aria-describedby', this.id);
-			}
-			if (logAccessibilityWarning && !isInteractive && !this.announced) {
-				console.warn(
-					'd2l-tooltip may be being used in a non-accessible manner; it should be attached to interactive elements like \'a\', \'button\',' +
-					'\'input\'', '\'select\', \'textarea\' or static / custom elements if a role has been set and the element is focusable.',
+
+			if (this.#targetDelegated) {
+				if (this.forType === 'label') {
+					this.removeAttribute('aria-labelledby');
+					this._target.ariaLabel = this.shadowRoot.querySelector('slot').assignedNodes().map(n => n.textContent).join('\n');
+				} else if (!this.announced || isInteractive) {
+					this.removeAttribute('aria-describedby');
+					this._target.ariaDescription = this.shadowRoot.querySelector('slot').assignedNodes().map(n => n.textContent).join('\n');
+				}
```

Comment:
When the target has been delegated, we can't use the `...by` properties, which take precedence, so remove them.

Reply:
Whoops.
```suggestion
				if (this.forType === 'label') {
					this._target.removeAttribute('aria-labelledby');
					this._target.ariaLabel = this.shadowRoot.querySelector('slot').assignedNodes().map(n => n.textContent).join('\n');
				} else if (!this.announced || isInteractive) {
					this._target.removeAttribute('aria-describedby');
					this._target.ariaDescription = this.shadowRoot.querySelector('slot').assignedNodes().map(n => n.textContent).join('\n');
				}
```